lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -543,13 +543,20 @@ def get_distribution_SunOS(self):
         data = get_file_content('/etc/release').splitlines()[0]
 
         if 'Solaris' in data:
+            # FIXME: the proper way would be to exec 'uname -r' and parse it.
+            # For solaris 10 it will return 5.10, for solaris 11 it will be 5.11
+            # but I can't pass ansible test unit test_distribution_version.py with it,
+            # since I've no idea how to mock `uname -r` on CI (github/shippable) run.
+            # rc, uname_r, err = self.module.run_command('uname -r')
```

Comment:
cc @mattclay

Reply:
Was there something in particular you wanted me to look at?

Reply:
Ah, let me see if I can find an example of mocking run_command.

Reply:
We already have a `get_uname_version` function:

https://github.com/ansible/ansible/blob/84611ebc5ad1dbbfa589ef257b343bf88f5287b6/lib/ansible/module_utils/facts/system/distribution.py#L28-L32

A `get_uname_release` function could be added to match and then mocked the same way:

https://github.com/ansible/ansible/blob/84611ebc5ad1dbbfa589ef257b343bf88f5287b6/test/units/module_utils/test_distribution_version.py#L997